inc/Engine/CriticalPath/RESTWP.php

```diff
@@ -100,7 +110,39 @@ public function check_permissions() {
 	 * @return WP_REST_Response
 	 */
 	public function generate( WP_REST_Request $request ) {
-		$item_id = (int) $request->get_param( 'id' );
+		$item_id   = (int) $request->get_param( 'id' );
+		$is_mobile = (bool) $request->get_param( 'is_mobile' );
+
+		// Bailout in case mobile CPCSS generation is called but this option is disabled.
+		if ( $is_mobile && ! $this->options->get( 'async_css_mobile', 0 ) ) {
+			return rest_ensure_response(
+				$this->return_error(
+					new WP_Error(
+						'mobile_cpcss_not_enabled',
+						__( 'Mobile CPCSS generation not enabled.', 'rocket' ),
+						[
+							'status' => 400,
+						]
+					)
+				)
+			);
+		}
+
+		// Bailout in case separate cache for mobile option is enabled.
+		if ( $is_mobile && $this->options->get( 'do_caching_mobile_files', 0 ) ) {
+			return rest_ensure_response(
+				$this->return_error(
+					new WP_Error(
+						'caching_mobile_separate_enabled',
+						__( 'Separate cache files for mobile devices option is enabled.', 'rocket' ),
+						[
+							'status' => 400,
+						]
+					)
+				)
+			);
+		}
```

Comment:
I don't understand this part, is there something missing?

Reply:
@Tabrisrp It's part of this commit https://github.com/wp-media/wp-rocket/commit/3a7adbfc5607f65455d91d981465a9fcfc94dbd6. What should this be?

Reply:
it should be the opposite, we should prevent generation if the option is disabled

Reply:
https://github.com/wp-media/wp-rocket/pull/2672/commits/726e28ac72a6d497cfec4e09e0f98e9059a5e939